### PR TITLE
Mining Master Support

### PR DESCRIPTION
**Pull request by SiverDX**
https://www.curseforge.com/minecraft/mc-mods/mining-master

https://github.com/infernalstudios/Mining-Master/tree/1.19/src/main/resources/data/miningmaster/worldgen/placed_feature

image

image

* No Malachite right now due to the way it generates
* Need to add Deepslate variants

### CODE REVIEW

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import net.minecraft.world.item.ItemStack;
+import org.infernalstudios.miningmaster.init.MMBlocks;
+import org.infernalstudios.miningmaster.init.MMItems;
+import org.jetbrains.annotations.NotNull;
+
+public class MiningMasterWorldGen extends WorldGenIntegration {
+    @Override
+    public void registerWorldGen(@NotNull final IWorldGenRegistry registry) {
+        registerStoneType(registry);
+        registerDeepslateType(registry);
```
Comment (AlasDiablo) [inline src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java]: Please use `this`

```java
this.registerStoneType(registry);
this.registerDeepslateType(registry);
```

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+        registry.register(
+                new ItemStack(MMBlocks.POWER_PYRITE_ORE.get()),
+                new DistributionTriangular(1, 1, 64, 64),
+                new Restriction(DimensionRestriction.NETHER),
+                true,
+                new LootDrop(new ItemStack(MMItems.POWER_PYRITE.get()), 1, 1, Conditional.affectedByFortune)
+        );
+    }
+    private void registerStoneType(@NotNull final IWorldGenRegistry registry) {
```
Comment (AlasDiablo) [inline src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java:16]: This function is call by `registerWorldGen`, i highly recommend to put it before `registerWorldGen`.

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        registry.register(
+                new ItemStack(MMBlocks.UNBREAKING_IOLITE_ORE.get()),
+                new DistributionTriangular(2, 1, -10, 30),
+                new Restriction(DimensionRestriction.OVERWORLD),
+                true,
+                new LootDrop(new ItemStack(MMItems.UNBREAKING_IOLITE.get()), 1, 1, Conditional.affectedByFortune)
+        );
+    }
+
+    private void registerDeepslateType(@NotNull final IWorldGenRegistry registry) {
```
Comment (AlasDiablo) [inline src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java:90]: This function is call by `registerWorldGen`, i highly recommend to put it before `registerWorldGen`.

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+public class MiningMasterWorldGen extends WorldGenIntegration {
+    @Override
+    public void registerWorldGen(@NotNull final IWorldGenRegistry registry) {
+        registerStoneType(registry);
+        registerDeepslateType(registry);
+
+        registry.register(
+                new ItemStack(MMBlocks.HEART_RHODONITE_ORE.get()),
+                new DistributionTriangular(1, 1, 64, 64),
```
Comment (AlasDiablo) [inline src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java]: Mining Master did have a config system, or is hard coded ?

**Review by AlasDiablo — CHANGES_REQUESTED** [review]
Hi, thank.

This pr need some change, after that i will look to make a release.

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java (hunk 3 above)
Comment (SiverDX) [reply src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java:90]: Do you mean to put the method above the `registerWorldGen` method (= refactor) or something else?

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java (hunk 4 above)
Comment (SiverDX) [reply src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java]: They do not generate a config

In theory you could configure everything with datapacks

Maybe for the future there could be a system where you read the `PlacedFeature` resource as json or sth. to get the actual values dynamically

Comment (AlasDiablo) [reply src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java]: Yep, my other mod Janoeo Foundation have now config file, but i need to check who to get access to those file durring runtime.

File: src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java (hunk 3 above)
Comment (AlasDiablo) [reply src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java:90]: Yes, is "better" to have method call in a function declared above the function who use it.

Comment (SiverDX) [reply src/main/java/fr/alasdiablo/jerintegration/compat/miningmaster/MiningMasterWorldGen.java:90]: Made the change but are you sure you want it that way?
I moved the types to their own methods to make `registerWorldGen` more readable but now it's pretty far down

**Review by AlasDiablo — APPROVED** [review]
I will merge this, later this day, thank.

**Pull request merged**